test/unittest/tests/block/test_block_api_misc.cpp

```diff
@@ -0,0 +1,242 @@
+    CHECK(static_cast<std::string>(((char *)(buf.data))).compare(expected_str) == 0);
+
+    __wt_free(session, buf.data);
+}
+
+// Tested using a white-box approach as we need knowledge of internal structures to test various
+// inputs.
+TEST_CASE("Block manager addr string", "[block_api_misc]")
+{
+    std::shared_ptr<mock_session> session;
+
+    WT_BM bm;
+    WT_CLEAR(bm);
+    initialize_bm(session, &bm);
+    WT_SESSION_IMPL *s = session->get_wt_session_impl();
+
+    SECTION("Test addr string with non-zero values")
+    {
+        // (512, 1024, 12345) -> (offset, size, checksum)
+        test_addr_string(s, &bm, 512, 1024, 12345, "[0: 512-1536, 1024, 12345]");
+    }
+
+    SECTION("Test addr string with zero values")
+    {
+        // (0, 0, 0) -> (offset, size, checksum)
+        test_addr_string(s, &bm, 0, 0, 0, "[0: 0-0, 0, 0]");
+    }
+
+    SECTION("Test addr string with zero size")
+    {
+        // (512, 0, 12345) -> (offset, size, checksum)
+        test_addr_string(s, &bm, 512, 0, 12345, "[0: 0-0, 0, 0]");
+    }
+
+    REQUIRE(__wt_block_close(s, bm.block) == 0);
+}
+
+// Tested using a black-box approach.
+TEST_CASE("Block header", "[block_api_misc]")
+{
+    // Declare a block manager and set it up so that we can use its legal API methods.
+    WT_BM bm;
+    WT_CLEAR(bm);
+    __ut_bm_method_set(&bm);
+
+    REQUIRE(bm.block_header(&bm) == (u_int)WT_BLOCK_HEADER_SIZE);
+}
+
+// Tested using a white-box approach as using a black-box approach would be pretty involved for the
+// purposes of this test.
+TEST_CASE("Block manager is mapped", "[block_api_misc]")
+{
+    // Declare a block manager and set it up so that we can use its legal API methods.
+    WT_BM bm;
+    WT_CLEAR(bm);
+    __ut_bm_method_set(&bm);
+
+    SECTION("Test block manager is mapped")
+    {
+        uint8_t i;
+        bm.map = &i;
+        REQUIRE(bm.is_mapped(&bm, nullptr) == true);
+    }
+
+    SECTION("Test block manager is not mapped")
+    {
+        bm.map = nullptr;
+        REQUIRE(bm.is_mapped(&bm, nullptr) == false);
+    }
+}
+
+TEST_CASE("Block manager size and stat", "[block_api_misc]")
+{
+    std::shared_ptr<mock_session> session;
+
+    WT_BM bm;
+    WT_CLEAR(bm);
+    initialize_bm(session, &bm);
+    WT_SESSION_IMPL *s = session->get_wt_session_impl();
+
+    // Test that the bm->stat method updates statistics correctly after initializing the block
+    // manager.
+    check_bm_stats(s, &bm);
+
+    // Perform a write.
+    WT_ITEM buf;
+    WT_CLEAR(buf);
+    std::string test_string("test123");
+    create_write_buffer(&bm, session, test_string, &buf, 0);
+    uint8_t addr[WT_BTREE_MAX_ADDR_COOKIE];
+    size_t addr_size;
+    wt_off_t bm_size;
+    REQUIRE(bm.write(&bm, s, &buf, addr, &addr_size, false, false) == 0);
+    REQUIRE(bm.size(&bm, s, &bm_size) == 0);
+
+    // Test that the bm->data method updates statistics correctly after doing a write.
+    check_bm_stats(s, &bm);
```

Comment:
do the stats not change after the write?

Reply:
After doing a write, the block size stat increases.